File: Backend/src/main/java/in/Sumit/foodiesapi/service/CartService.java
package in.Sumit.foodiesapi.service;

import in.Sumit.foodiesapi.io.CartRequest;
import in.Sumit.foodiesapi.io.CartResponse;

public interface CartService {

    CartResponse addToCart(CartRequest request);

    CartResponse getCart();

    void clearCart();

    CartResponse removeFromCart(CartRequest cartRequest);
}

File: Frontend/src/services/cartService.ts
import api from '../config/api';
import { Cart } from '../types';

export const cartService = {
  async getCart(): Promise<Cart> {
    try {
      const response = await api.get('/cart');
      return response.data;
    } catch (error: any) {
      throw error;
    }
  },

  async addToCart(foodId: string): Promise<Cart> {
    try {
      const response = await api.post('/cart', { foodId });
      return response.data;
    } catch (error: any) {
      throw error;
    }
  },

  async removeFromCart(foodId: string): Promise<Cart> {
    try {
      const response = await api.post('/cart/remove', { foodId });
      return response.data;
    } catch (error: any) {
      throw error;
    }
  },

  async clearCart(): Promise<void> {
    try {
      await api.delete('/cart');
    } catch (error: any) {
      throw error;
    }
  }
};